server/services/discovery/discovery_service.py

```
@@ -333,4 +405,12 @@ def run_discovery_for_user(user_id, connected_providers):
         f"{total_nodes} nodes, {total_edges} edges, "
         f"{len(summary['errors'])} errors, {elapsed:.1f}s"
     )
+
+    # Clean up any ephemeral gcloud temp directories created during this run.
+    import shutil
+    for _pname, (env, _) in provider_envs.items():
```

Ruff (0.15.12)

[warning] When using only the values of a dict use the `values()` method

Replace `.items()` with `.values()`

(PERF102)